hack/licence-detector/main.go

```diff
@@ -27,6 +28,7 @@ var (
 	noticeOutFlag       = flag.String("noticeOut", "NOTICE.txt", "Path to output the notice")
 	overridesFlag       = flag.String("overrides", "", "Path to the file containing override directives")
 	validateFlag        = flag.Bool("validate", false, "Validate results (slow)")
+	whitelistFlag       = flag.Bool("checkWhitelist", true, "Checks that all dependencies are whitelisted, returns an error if any are not")
```

Comment:
I wonder whether the whitelist check should just be a part of the validation check.

Reply:
The whitelist check is much faster so it seems to make sense to run it by default on every build.

Reply:
Then perhaps we could move the check to the licence detection stage and remove the extra loop here. It's a pretty minor optimisation + stylistic preference though. I don't feel too strongly about it.

Reply:
Nah you're right, this doesn't really need to be behind a flag and makes it much smaller